### DISCUSSION TITLE
Sub Editor - Сan’t save the sub immediately (Shadow light limit)

### DISCUSSION THREAD

**Discussion by MaaiG** [FakeFishGames/Barotrauma#9308] [Ideas and suggestions]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

Can’t save the sub if the number of `shadow lights` is over 60.
It is necessary to save the result and quit the game urgently but you can't do it. You must change the settings of excess lamps!
You can use "Esc" and auto-save file but this is not the best option.
![Image 001_13-58](https://user-images.githubusercontent.com/93348499/173051458-a16517ab-3af8-4a0a-82d5-8bd5f5075188.png)

### Reproduction steps

1. Start the editor
2. Build something long and hard :)
3. Number of shadow lamps more than 60 (much more)
4. You need to urgently finish the job and get out of the game
5. You can save your work the usual way!

### Bug prevalence

Happens every time I play

### Version

0.17.16.0

### Which operating system did you encounter this bug on?

Windows

**Comment by Regalis11** [comment]
This is not a bug, I'll convert this into a discussion

**Reply by JoneKone** [reply]
Why does the Auto save work? **Also asked for a feature to filter only shadow casting lights, ages ago**. now you make this error and give no option to even search them.

**Reply by MaaiG** [reply]
> ...no option to even search them.

That was my first thought, too.
Now there is a function "Select matching items" (Shift + RMB) but it does not distinguish between lights and shadow casting lights!
![Image 001_15-53](https://user-images.githubusercontent.com/93348499/174074276-9aae4770-3af0-4ac1-a8f4-23be466f991a.png)

**Comment by MaaiG** [comment]
Technically, yes! Apparently it’s intentionally! This is not a bug but I can’t keep my job! I guess it’s a mistake!
I can’t finish the submarine in one session! I want to leave many solutions to the next time!

And furthermore, there were no such restrictions in previous versions!
1. I open my old submarine (Shadow light >60)
2. Change it
3. I try to save and exit
4. Get such a surprise! :)
The first time I didn’t pay attention to this window and went out. My work did not survive! :(

For example:
Currently, there are no directional light sources without a shadow in the game. All lamps emit light in all directions. It takes a lot of effort to make a minimum number of shadow lamps and provide reasonable lighting design.
It would be easier if lamps without shadows could emit adjustable directional light:
![Light](https://user-images.githubusercontent.com/93348499/173058525-174d9ef1-183a-4b96-8baa-8a59965c5aab.gif)
[This could simplify and speed up the process of placing light sources without shadows...](https://github.com/Regalis11/Barotrauma/discussions/8052)
But now I can’t quickly adjust all the light sources I want to leave it for the future. I need to Save and Get out! Now I can’t do it! I’ll lose all my work!

**Comment by MaaiG** [comment]
We have "duct tape" - https://github.com/Regalis11/Barotrauma/issues/9364#issuecomment-1157523415 😛

**Reply by JoneKone** [reply]
Btw I don't want too many shadow casting lights.. I reduced it to 50 in the ship that had over 70

**Reply by MaaiG** [reply]
You have a large submarine! Some rooms do not illuminate?
If I turn off the shadows light then the light from the lamp lights up the other rooms!
I can change the lamp range, but not all the lamps are in the middle of the room and many rooms are long and rectangular. Need a directional light source without shadows light.

**Reply by JoneKone** [reply]
I add more lights. smaller lights to compensate. Also you can use the spot light,

**Reply by MaaiG** [reply]
Oh, I try :)
![Image 001_11-47](https://user-images.githubusercontent.com/93348499/174430306-002f4987-b18a-4aac-a40b-c73a6cdec078.png)
